### ISSUE TITLE
bug: Unknown source position when using lambda in inner class

### ISSUE THREAD

**Issue by aveuiller** [INRIA/spoon#3052]
Hello,

I have a quite weird behavior in my processor when analyzing this class: https://github.com/woffs/tazapp-android/blob/master/tazapp/src/main/java/de/thecode/android/tazreader/reader/usertoc/UserTocAdapter.java#L180-L191

I tracked down the bug to this minimal code sample: 
```java
public class Test {
    public static class Failing {
        Any listener;
    }

    private static class ExtendedClass extends Failing {
        public void test() { 
           // The difference in behaviour happens if I comment this line out
           Runnable r = v -> listener.someAction();
        }
    }
}
```

Here, the class `Failing` will have a position returning `(uknown file)` in my processor if the line with the lambda stays uncommented.
My processor will work fine if I comment it out though.

You will find as follow the configuration of my spoon launcher:
```java
        Launcher launcher = new Launcher();

        launcher.addInputResource(appPath);
        launcher.getEnvironment().setNoClasspath(true);
        launcher.buildModel();

        AbstractProcessor<CtClass> classProcessor = new ClassProcessor();
        AbstractProcessor<CtInterface> interfaceProcessor = new InterfaceProcessor();
        launcher.addProcessor(classProcessor);
        launcher.addProcessor(interfaceProcessor);
        launcher.process();
```

And I will also attach the two states from my debugger.
Do you have an idea of the issue? Anything I can do about it?

Thanks in advance.

![2019-07-18-233325_565x504_scrot](https://user-images.githubusercontent.com/4600261/61494245-f40a4800-a9b5-11e9-9fa4-9fa8ef9d7141.png)
![2019-07-18-233420_1045x419_scrot](https://user-images.githubusercontent.com/4600261/61494246-f4a2de80-a9b5-11e9-9ff1-b749a7a05a16.png)

**Comment by monperrus** [comment]
Good to see you Antoine.

AFAIU, your problem is that you have an empty position?

**Comment by aveuiller** [comment]
Hello Martin,

You are correct. In my processor, I am using the `position` attribute to store the path to the file holding the processed class, but in this case I am not able to retrieve it.

By the way, I forgot to mention but I'm using the version `7.4.0`.

**Comment by monperrus** [comment]
So this is a bug. Would you create a pull-request with a failing test case? Thanks!

**Comment by aveuiller** [comment]
A reproduction of the bug is available in #3054
